### ISSUE TITLE
Normalizing the layer weights in loss function has no effect

### ISSUE THREAD

**Issue by ehalit** [cvqluu/Additive-Margin-Softmax-Loss-Pytorch#13]
Hello!

The parameters of the fully connected layer in the loss function are normalized in the following way:

```
for W in self.fc.parameters():
    W = F.normalize(W, p=2, dim=1)
```

However, the weights of `self.fc` are not effected from this operation, I checked it with `print(self.fc.weight)` and `print(W)`. This means that the cosine calculation is actually conducted with non-normalized vectors.

**Comment by ehalit** [comment]
This seems to be a duplicate issue, see the [original one](https://github.com/cvqluu/Angular-Penalty-Softmax-Losses-Pytorch/issues/3).
